kafka-connect-http/src/main/scala/io/lenses/streamreactor/connect/http/sink/HttpWriterManager.scala

```diff
@@ -0,0 +1,202 @@
+          config.authentication,
+          config.method.toHttp4sMethod,
+          cRes,
+        )
+        new HttpWriterManager(
+          sinkName,
+          template,
+          requestSender,
+          config.batch.map(_.toCommitPolicy).getOrElse(HttpCommitPolicy.Default),
+          cResRel,
+          Ref.unsafe(Map[Topic, HttpWriter]()),
+          config.errorThreshold.getOrElse(DefaultErrorThreshold),
+          config.uploadSyncPeriod.getOrElse(DefaultUploadSyncPeriod),
+        )
+    }
+
+  }
+}
+class HttpWriterManager(
+  sinkName:          String,
+  template:          TemplateType,
+  httpRequestSender: HttpRequestSender,
+  commitPolicy:      CommitPolicy,
+  val close:         IO[Unit],
+  writersRef:        Ref[IO, Map[Topic, HttpWriter]],
+  errorThreshold:    Int,
+  uploadSyncPeriod:  Int,
+) extends LazyLogging {
+  private val scheduler: ScheduledExecutorService = Executors.newSingleThreadScheduledExecutor()
+
+  private def createNewHttpWriter(): HttpWriter =
+    new HttpWriter(
+      sinkName     = sinkName,
+      commitPolicy = commitPolicy,
+      sender       = httpRequestSender,
+      template     = template,
+      Ref.unsafe[IO, Queue[RenderedRecord]](Queue()),
+      Ref.unsafe[IO, HttpCommitContext](HttpCommitContext.default(sinkName)),
+      errorThreshold,
+    )
+
+  def getWriter(topic: Topic): IO[HttpWriter] = {
+    var foundWriter = Option.empty[HttpWriter]
+    for {
+      _ <- writersRef.getAndUpdate {
+        writers =>
+          foundWriter = writers.get(topic)
+          if (foundWriter.nonEmpty) {
+            writers // no update
+          } else {
+            val newWriter = createNewHttpWriter()
+            foundWriter = newWriter.some
+            writers + (topic -> newWriter)
+          }
+      }
+      o <- IO.fromOption(foundWriter)(new IllegalStateException("No writer found"))
+    } yield o
+  }
+
+  // answers the question: what have you committed?
+  def preCommit(currentOffsets: Map[TopicPartition, OffsetAndMetadata]): IO[Map[TopicPartition, OffsetAndMetadata]] = {
+
+    val currentOffsetsGroupedIO: IO[Map[Topic, Map[TopicPartition, OffsetAndMetadata]]] = IO
+      .pure(currentOffsets)
+      .map(_.groupBy {
+        case (TopicPartition(topic, _), _) => topic
+      })
+
+    for {
+      curr    <- currentOffsetsGroupedIO
+      writers <- writersRef.get
+      res <- writers.toList.traverse {
+        case (topic, writer) =>
+          writer.preCommit(curr(topic))
+      }.map(_.flatten.toMap)
+    } yield res
+  }
+
+  def start(errCallback: Throwable => Unit): IO[Unit] =
+    for {
+      _ <- IO(logger.info(s"[$sinkName] starting HttpWriterManager"))
+      runnable: Runnable = () => {
+        process().flatMap {
+          resultList =>
+            handleResult(resultList, errCallback)
+        }.unsafeRunSync()
+        ()
+      }
+      _ <- IO.delay(scheduler.scheduleAtFixedRate(runnable, 0, uploadSyncPeriod.toLong, TimeUnit.SECONDS))
```

Comment:
Let's have a look at this on Monday maybe. 

Unfortunately, doing it like this means that it will be ever running and cannot be stopped. 
Running the test from `HttpSinkTaskIT` you can see that even tho the test terminates this process keeps running until the jvm is terminated. Ideally we would want to have a "stop" method that could interrupt this scheduling.